examples/imagemagick_globus.py

```diff
@@ -50,10 +47,13 @@ def main():
         override_existing=False,
     )
 
-    campaign.add_activity(activity)
-    campaign.add_activity(transfer)
+    # create campaign and dispatch it
+    campaign = Campaign(
```

Comment:
I think switching from adding activities as its own function  to part of the Campaign instantiation (as you did) is cleaner. And at this point I believe this is okay functionally, but soon we will have to have dynamically-added tasks that might require the campaign having more activities added as the workflow is executed. So we may have to take slightly more care to ensure that the activity object in the Campaign is mutable.

Reply:
The `Campaign` class has an `add_activity` method to add more activities. So this should be able to handle dynamic tasks but we'll need to develop some examples and tests to see what the limitations are.